dist/angular-spring-data-rest.js

```diff
@@ -250,6 +252,10 @@ angular.module("spring-data-rest").provider("SpringDataRestAdapter", function ()
 
                         // if there are links to fetch, then process and fetch them
                         if (fetchLinkNames != undefined) {
+                            var self = data._links.self.href;
```

Comment:
Please use this as `self` variable:

```
var self = data[config.linksSelfLinkName];
```

Then we assure that always the proper `self` link is taken.